sycl/include/sycl/info/aspects.def

```diff
@@ -0,0 +1,38 @@
+  __SYCL_ASPECT(host, 0, "host")
+  __SYCL_ASPECT(cpu, 1, "cpu")
+  __SYCL_ASPECT(gpu, 2, "gpu")
+  __SYCL_ASPECT(accelerator, 3, "accelerator")
+  __SYCL_ASPECT(custom, 4, "custom")
+  __SYCL_ASPECT(fp16, 5, "fp16")
+  __SYCL_ASPECT(fp64, 6, "fp64")
+  __SYCL_ASPECT(int64_base_atomics __SYCL2020_DEPRECATED("use atomic64 instead"), 7, "int64_base_atomics")
+  __SYCL_ASPECT(int64_extended_atomics __SYCL2020_DEPRECATED("use atomic64 instead"), 8, "int64_extended_atomics")
+  __SYCL_ASPECT(image, 9, "image")
+  __SYCL_ASPECT(online_compiler, 10, "online_compiler")
+  __SYCL_ASPECT(online_linker, 11, "online_linker")
+  __SYCL_ASPECT(queue_profiling, 12, "queue_profiling")
+  __SYCL_ASPECT(usm_device_allocations, 13, "usm_device_allocations")
+  __SYCL_ASPECT(usm_host_allocations, 14, "usm_host_allocations")
+  __SYCL_ASPECT(usm_shared_allocations, 15, "usm_shared_allocations")
+  __SYCL_ASPECT(usm_restricted_shared_allocations, 16, "usm_restricted_shared_allocations")
+  __SYCL_ASPECT(usm_system_allocations, 17, "usm_system_allocations")
+  __SYCL_SPECIAL_ASPECT(usm_system_allocator __SYCL2020_DEPRECATED("use usm_system_allocations instead"), usm_system_allocations, "usm_system_allocations")
```

Comment:
Can we have or a more precise definition of "special"? Also, I'd probably move deprecation message into a separate macro argument and call it

```
__SYCL_ASPECT_DEPRECATED(ASPECT, ID, MESSAGE)
```

Reply:
1. Added the comment to describe this macro, does it look better now?
2. Added `__SYCL_ASPECT_DEPRECATED` macro & outlined these aspects into the separate file. https://github.com/intel/llvm/pull/6989/commits/410f00eabc04a1ab5be42dc4da11c20fd07d51ae